### ISSUE TITLE
Unable to download version 2.20.3

### ISSUE THREAD

**Issue by anfperez** [hashicorp/terraform-provider-google#12255]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```
Terraform v1.2.0
on darwin_amd64
+ provider registry.terraform.io/datadog/datadog v2.26.1
+ provider registry.terraform.io/hashicorp/google v2.20.3
+ provider registry.terraform.io/hashicorp/google-beta v4.27.0
```

### Affected Resource(s)

```
registry.terraform.io/hashicorp/google:
google_2.20.3
* google_XXXXX
```

### Terraform Configuration Files

```tf
terraform {
  required_providers {
    datadog = {
      source  = "datadog/datadog"
      version = "~> 2.12"
    }
    google = {
      source  = "hashicorp/google"
      version = "~> 2.20"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 4.27.0"
    }
  }
  required_version = ">= 0.13"

  backend "gcs" {
  }
}
```

### Debug Output

https://gist.github.com/anfperez/80c2cbb1c17dc6b411726db3fc61588b

### Panic Output

none

### Expected Behavior

Terraform should download the Google plugin at version 2.20.3

### Actual Behavior

```
│ Error: Failed to install provider
Partner and community providers are signed by their developers.
│
If you'd like to know more about provider signing, you can read about it here:
│ Error while installing hashicorp/google v2.20.3: could not query provider
https://www.terraform.io/docs/cli/plugins/signing.html
│ registry for registry.terraform.io/hashicorp/google: registry response
│ includes invalid version string "4,5": can't specify multiple versions; a
│ single exact version is required
╵
```

### Steps to Reproduce

`terraform init` with a Google provider of 2.20.3

### Important Factoids

```
curl 'https://registry.terraform.io/v1/providers/hashicorp/google/versions' | jq | grep "4,5"
        "4,5"
```

**Comment by SarahFrench** [comment]
Hi @anfperez , thanks for raising this problem!

It looks like that version of the provider needs `{"protocols":["4,5"],` changed to `{"protocols":["4.0","5.0"],` in the Terraform Registry's data about that release. I've asked the Registry team to confirm that it's a bug (v2.20.3 is about 3 years old and the release process has changed since then) and let me know if that data can be edited by them.

I'll report back here with what they say.

In the meantime, a workaround could be to download the provider from the registry yourself ([download URL](https://releases.hashicorp.com/terraform-provider-google/2.20.3/terraform-provider-google_2.20.3_linux_amd64.zip) is shown [here](https://registry.terraform.io/v1/providers/hashicorp/google/2.20.3/download/linux/amd64)) and configure Terraform to use a [filesystem mirror](https://www.terraform.io/cli/config/config-file#explicit-installation-method-configuration) specifically for that provider

**Comment by SarahFrench** [comment]
@anfperez Could you please check if you're able to download the provider?

The Registry team has fixed the issue I described and I was able to download that provider version with the configuration you posted 🎉

**Comment by abdrehma** [comment]
Hi just came across this - error still shows:

```
Initializing provider plugins...
- Finding hashicorp/google versions matching "~> 2.20"...
- Finding hashicorp/google-beta versions matching "~> 4.27.0"...
- Installing hashicorp/google-beta v4.27.0...
- Installed hashicorp/google-beta v4.27.0 (signed by HashiCorp)

Error: Failed to install provider

Error while installing hashicorp/google v2.20.3: could not query provider
registry for registry.terraform.io/hashicorp/google: registry response
includes invalid version string "4,5": can't specify multiple versions; a
single exact version is required
```

**Comment by anfperez** [comment]
@SarahFrench, unfortunately, I am still getting the same error as well. But we might just need to wait for the update to propagate, so I will try again on Monday.

**Comment by ragaskar** [comment]
I am also seeing this error.

**Comment by yitzchake** [comment]
Error while installing hashicorp/google-beta v2.20.3: could not query
│ provider registry for registry.terraform.io/hashicorp/google-beta: registry
│ response includes invalid version string "4,5": can't specify multiple
│ versions; a single exact version is required

**Comment by SarahFrench** [comment]
Hi all, I'm waiting on help from the Registry team- the [API response that lists all versions](https://registry.terraform.io/v1/providers/hashicorp/google/versions) how includes `{"version":"2.20.3","protocols":["4.0","5.0"],` but in the data for [specific builds of that version](https://registry.terraform.io/v1/providers/hashicorp/google/2.20.3/download/linux/amd64) it still shows the old `"protocols":["4,5"],`.

Hopefully once that difference is sorted it will solve the issue 

I'm not sure why I was able to download the provider successfully, but I am using darwin instead of linux so behaviour could vary

**Comment by SarahFrench** [comment]
After a bunch of cache purging, the fix should now be available for all os/arch builds of this version - @anfperez can you do `terraform init` ok now?

**Comment by tolgaio** [comment]
yup - it's working for me now after the latest update! Thanks.

**Comment by SarahFrench** [comment]
Brilliant, in that case I'll mark the issue as closed - thanks for everyone's patience!

**Comment by SarahFrench** [comment]
Note: this issue has returned and a new issue is opened here https://github.com/hashicorp/terraform-provider-google/issues/13859
